src/main/flight/servos.c

```diff
@@ -234,6 +237,29 @@ void servosInit(void)
         }
 
     }
+
+    for (uint8_t i = 0; i < MAX_SUPPORTED_SERVOS; i++) {
+        servoMetadata[i].rateSum = 0;
+        servoMetadata[i].scaleMax = 0;
+        servoMetadata[i].scaleMin = 0;
+    }
+
+    /*
+     * Iterate over mixer rules to determine max. possible rate for a servo
+     */ 
+    for (uint8_t i = 0; i < servoRuleCount; i++) {
+        servoMetadata[currentServoMixer[i].targetChannel].rateSum += abs(currentServoMixer[i].rate);
+    }
+
+    /*
+     * Compute scaling factor for upper and lower servo throw
+     */ 
+    for (uint8_t i = 0; i < MAX_SUPPORTED_SERVOS; i++) {
+        const float rateUs = servoMetadata[i].rateSum / 100.0f * 500;
```

Comment:
yes, this is the case. So we know max theoretical value that can be written to servo and scale from this value to max/min value we set for servo